Fix for flexbox bug with tooltip when maxWidth is set manually.

jdmarshall opened the pull request:
fix for #105517

This PR corrects an issue with maxWidth:undefined encountered in #105805 

Labels now grow unless maxWidth is set

<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

7. If your PR content should be added to the What's New document for the next major or minor release, add the **add to what's new** label to your PR. Note that you should add this label to the main PR that introduces the feature; do not add this label to smaller PRs for the feature.

-->

**What is this feature?**

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

jdmarshall committed:
Fix for flexbox bug with tooltip when maxWidth is set manually.

Retains full width if maxWidth is not set.

fix for #105517

jdmarshall commented:
Reviewer is advised to verify that tooltips with long labels are unadulterated when max width is left unset.

jdmarshall commented:
Could someone take a look at this?

jesdavpet reviewed (commented):
Hi @jdmarshall, thank you for contributing this bug fix to Grafana! 😄🙌

Sorry the 1st attempt had to be reverted. I feel like this one is really close, and it has avoided the problem with https://github.com/grafana/grafana/pull/105805.

When I tried these changes out locally, I noticed one minor bug that will need to be fixed, but everything else LGTM so far.

----

**bug(blocking):** 🐞 It's important to retain the colour coding of the series icons (legend colours, left column), there's a minor issue in this PR so far that causes them to disappear for certain `Max width`s.

Starting at a`Max width` of around `750`, the series icons start to get squished horizontally, and eventually disappear at smaller `Max width` sizes.

Squished series icons pictured below:
<img width="803" height="468" alt="Screenshot 2025-07-21 at 3 49 25 PM" src="https://github.com/user-attachments/assets/1c35ab67-6645-4ab5-be52-0d4fe79b5c86" />


Correctly sized series icons pictured below:
<img width="1527" height="741" alt="Screenshot 2025-07-21 at 3 58 24 PM" src="https://github.com/user-attachments/assets/609fc6ad-f25f-453d-a33b-b8658d2bdad2" />

It's OK if the series icons shrink a bit horizontally at very small tooltip sizes, but their width should never get any narrower horizontally than the vertical height, and it should never disappear. Another way to say it is, it would be OK if they become narrow enough that they're circular, but no narrower.

jdmarshall committed:
Prevent seriesIcon from shrinking away to nothing when in a flex-box
situation next to the label (eg, on tooltips).

This change has no effect on chart legends because they wrap instead
of shrink, so I think this is good.

jdmarshall commented:
Someone else has fixed this issue another way.
